### ISSUE TITLE
Firefox/VoiceOver: SVG `<title>`s within links are announced by VoiceOver but don't appear in VO's Web Rotor

### ISSUE THREAD

**Issue by querkmachine** [alphagov/reported-bugs#86]
**Upstream bug:** 
- Safari/WebKit: https://bugs.webkit.org/show_bug.cgi?id=265321
- Firefox/Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1866718

**Date:** 2023-11-24
**Reported by:** @querkmachine
**Related to:** https://github.com/alphagov/govuk-frontend/pull/4497

---

### Overview

When a link contains an inline SVG element, the `<title>` within the SVG will be announced by VoiceOver, but it will not appear in the link text displayed in VoiceOver's Web Rotor feature.

For example, in the following code:
```html
<a href="#">
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="42"
    height="42"
    viewBox="0 0 42 42"
  >
    <title>A red circle</title>
    <circle r="21" cx="21" cy="21" fill="red" />
  </svg>
  with accompanying text
</a>
```

### Expected result

- Navigating to the link via tabbing, VoiceOver keyboard controls, or the cursor will announce "A red circle with accompanying text".
- The links section of VoiceOver's Web Rotor will include an entry for "A red circle with accompanying text".
- Navigating to the link's entry in the Web Rotor will announce "A red circle with accompanying text". 

### Actual result

- Navigating to the link via tabbing, VoiceOver keyboard controls, or the cursor announces "A red circle with accompanying text".
- **The links section of VoiceOver's Web Rotor includes an entry for " with accompanying text".**
- Navigating to the link's entry in the Web Rotor announces "A red circle with accompanying text". 

### Other information

If the link only contains the SVG with no accompanying text, then the links `href` is displayed in the Web Rotor instead.

The issue can be mitigated by applying an `aria-label` with the same text as the `<title>`. Using `aria-labelledby` and an ID may also work. `aria-describedby` will not.

The same result was seen in Firefox 220, but not in Chrome 119, which displays the Web Rotor entries as expected.

The issue was not seen in the equivalent functionality available in JAWS 2023. I was unable to test NVDA 2023's version of the link listing.

**Comment by querkmachine** [comment]
This looks to have been resolved. Retesting it myself agrees it is!
